summary.sh

```diff
@@ -17,6 +17,10 @@ printf "\n\n\n RUN TESTS & TEST COVERAGE: \n"
 coverage run --source lisc  -m py.test
 coverage report
 
+# Check doctests - runs doctests with pytest, skipping normal tests
+printf "\n\n\nCHECK DOCTEST EXAMPLES: \n"
+pytest --doctest-modules --ignore=lisc/tests --ignore=lisc/data/utils.py lisc
```

Comment:
Why is the `data/utils` file skipped?

Reply:
I ignored lisc/data/utils.py because it seems that these tests are dependent on downloading nltk_data, yet there are not doctest examples here anyways. I had to open an interpreter and manually download the data to get these to pass. If someone where to try to run those tests without downloading nltk data, the error may be confusing. Nltk automatically looks for data in several locations, so you may have to clean those paths out to reproduce the error:

```
=========================================================== ERRORS ===========================================================
____________________________________________ ERROR collecting lisc/data/utils.py _____________________________________________
.env/lib/python3.8/site-packages/nltk/corpus/util.py:83: in __load
    root = nltk.data.find("{}/{}".format(self.subdir, zip_name)).env/lib/python3.8/site-packages/nltk/data.py:585: in find
    raise LookupError(resource_not_found)
E   LookupError: 
E   **********************************************************************
E     Resource stopwords not found.
E     Please use the NLTK Downloader to obtain the resource:
E   
E     >>> import nltk
E     >>> nltk.download('stopwords')
E     
E     For more information see: https://www.nltk.org/data.html
E   
E     Attempted to load corpora/stopwords.zip/stopwords/
E   
E     Searched in:E       - '/home/rph/nltk_data'
E       - '/home/rph/Projects/DSP/lisc/.env/nltk_data'
E       - '/home/rph/Projects/DSP/lisc/.env/share/nltk_data'
E       - '/home/rph/Projects/DSP/lisc/.env/lib/nltk_data'
E       - '/usr/share/nltk_data'
E       - '/usr/local/share/nltk_data'
E       - '/usr/lib/nltk_data'
E       - '/usr/local/lib/nltk_data'
E   **********************************************************************

During handling of the above exception, another exception occurred:
/usr/lib/python3.8/doctest.py:932: in find
    self._find(tests, obj, name, module, source_lines, globs, {})
.env/lib/python3.8/site-packages/_pytest/doctest.py:475: in _find
    doctest.DocTestFinder._find(  # type: ignore
/usr/lib/python3.8/doctest.py:991: in _find
    if ((inspect.isroutine(inspect.unwrap(val))
.env/lib/python3.8/site-packages/_pytest/doctest.py:427: in _mock_aware_unwrap
    return real_unwrap(obj, stop=_is_mocked)
/usr/lib/python3.8/inspect.py:520: in unwrap
    while _is_wrapper(func):
/usr/lib/python3.8/inspect.py:514: in _is_wrapper
    return hasattr(f, '__wrapped__') and not stop(f)
.env/lib/python3.8/site-packages/nltk/corpus/util.py:120: in __getattr__
    self.__load()
.env/lib/python3.8/site-packages/nltk/corpus/util.py:85: in __load
    raise e
.env/lib/python3.8/site-packages/nltk/corpus/util.py:80: in __load
    root = nltk.data.find("{}/{}".format(self.subdir, self.__name))
.env/lib/python3.8/site-packages/nltk/data.py:585: in find
    raise LookupError(resource_not_found)
E   LookupError: 
E   **********************************************************************
E     Resource stopwords not found.
E     Please use the NLTK Downloader to obtain the resource:
E   
E     >>> import nltk
E     >>> nltk.download('stopwords')
E     
E     For more information see: https://www.nltk.org/data.html
E   
E     Attempted to load corpora/stopwords
E   
E     Searched in:
E       - '/home/rph/nltk_data'
E       - '/home/rph/Projects/DSP/lisc/.env/nltk_data'
E       - '/home/rph/Projects/DSP/lisc/.env/share/nltk_data'
E       - '/home/rph/Projects/DSP/lisc/.env/lib/nltk_data'
E       - '/usr/share/nltk_data'
E       - '/usr/local/share/nltk_data'
E       - '/usr/lib/nltk_data'
E       - '/usr/local/lib/nltk_data'
E   **********************************************************************
====================================================== warnings summary ======================================================
.env/lib/python3.8/site-packages/_pytest/doctest.py:430
  /home/rph/Projects/DSP/lisc/.env/lib/python3.8/site-packages/_pytest/doctest.py:430: PytestWarning: Got LookupError("\n**********************************************************************\n  Resource \x1b[93mstopwords\x1b[0m not found.\n  Please use the NLTK Downloader to obtain the resource:\n\n  \x1b[31m>>> import nltk\n  >>> nltk.download('stopwords')\n  \x1b[0m\n  For more information see: https://www.nltk.org/data.html\n\n  Attempted to load \x1b[93mcorpora/stopwords\x1b[0m\n\n  Searched in:\n    - '/home/rph/nltk_data'\n    - '/home/rph/Projects/DSP/lisc/.env/nltk_data'\n    - '/home/rph/Projects/DSP/lisc/.env/share/nltk_data'\n    - '/home/rph/Projects/DSP/lisc/.env/lib/nltk_data'\n    - '/usr/share/nltk_data'\n    - '/usr/local/share/nltk_data'\n    - '/usr/lib/nltk_data'\n    - '/usr/local/lib/nltk_data'\n**********************************************************************\n") when unwrapping <WordListCorpusReader in '.../corpora/stopwords' (not loaded yet)>.  This is usually caused by a violation of Python's object protocol; see e.g. https://github.com/pytest-dev/pytest/issues/5080
    warnings.warn(

-- Docs: https://docs.pytest.org/en/latest/warnings.html
================================================== short test summary info ===================================================
ERROR lisc/data/utils.py - LookupError: 
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================ 1 warning, 1 error in 0.67s =================================================
```

Reply:
Oh. that is weird and annoying. The only file we use is the stopwords file. Maybe we just copy this file into the module, then we never have to be loading nltk files.

Reply:
We may have to edit the nltk dependency to change the default search path if we were to include the stopwords file into the module, since the traceback suggests the error only involves internal nltk code. 'punkt' is another nltk download that is required as well.

Reply:
Other LISC usage is also effect by not having the ntlk data downloaded. Running this without the nltk data downloaded in one of those designed locations:
```
articles = Articles('frontal lobe')
articles.load(SCDB('tutorials/lisc_db'))
articles_all = ArticlesAll(articles)
```
will raise an error:
```
LookupError: 
**********************************************************************
  Resource punkt not found.
  Please use the NLTK Downloader to obtain the resource:

  >>> import nltk
  >>> nltk.download('punkt')
  
  For more information see: https://www.nltk.org/data.html

  Attempted to load tokenizers/punkt/PY3/english.pickle

  Searched in:
    - '/home/rph/nltk_data'
    - '/home/rph/Projects/DSP/lisc/.env/nltk_data'
    - '/home/rph/Projects/DSP/lisc/.env/share/nltk_data'
    - '/home/rph/Projects/DSP/lisc/.env/lib/nltk_data'
    - '/usr/share/nltk_data'
    - '/usr/local/share/nltk_data'
    - '/usr/lib/nltk_data'
    - '/usr/local/lib/nltk_data'
    - ''
**********************************************************************
```

I'll see what the most efficient solution is to this. Maybe we should work on this in a separate issues/PR?

Reply:
Hmm yeh, that's annoying. This idea is sorta related to #2 - I had the idea of maybe switching the dependency, as I sorta don't like the big dependency of nltk. (I'll update the issue a little, and move discussion there).